R/metadata.R

```diff
@@ -118,6 +123,66 @@ get_barchart_settings <- function() {
   )
 }
 
+get_calibration_plot_settings <- function(filter_types) {
+  calibrate_only_settings <- list(
+    list(
+      filterId = scalar("calibrate_indicator"),
+      label = scalar(get_label_for_id("calibrate_indicator")),
+      stateFilterId = scalar("indicator")
+    )
+  )
+  filterIds <- c("spectrum_region", "type", "period", "sex", "age")
+  list(
+    defaultEffect = list(
+      setFilters = c(calibrate_only_settings,
+                     lapply(filterIds, get_filter_from_id)),
+      setFilterValues = list(
+        indicator = c("prevalence"),
+        period = get_filter_option_ids(filter_types, "period")[2],
+        sex = get_filter_option_ids(filter_types, "sex")[1],
+        age = c("Y015_049"),
+        calibrate_type = get_filter_option_ids(filter_types, "type"),
+        spectrum_region = get_filter_option_ids(filter_types, "spectrum_region")
+      ),
```

Comment:
is this necessary? the frontend will automatically pick everything if a filter is multiselect (which seems like what you want for `calibrate_type` and `spectrum_region`) and if it is single select then it will pick the first one, what you might be doing in `age` and `sex`, i think `indicator` and `period` are genuinely needed although not sure about why the current plot on production does that

if you do this then might be worth just hard coding the period id and no need to pass `filter_types` in, the options might potentially change and then youll have a different default without realising

Reply:
Oh great yes, I'll remove those unnecessary ones. I think we want to keep period working off the `filter_types` as these can change depending on the data. So we want the 2nd one chosen which should be the same as the time 2 option they select in the model options, and potentially different for different countries.